packages/plugin-ext/src/main/browser/view/tree-view-widget.tsx

```diff
@@ -128,7 +128,7 @@ export class PluginTree extends TreeImpl {
         try {
             const children = await proxy.$getChildren(this.identifier.id, parent.id);
             const oldEmpty = this._isEmpty;
-            this._isEmpty = !children || children.length === 0;
+            this._isEmpty = !parent.id && (!children || children.length === 0);
```

Comment:
Can you clarify why the `!parent.id` check is necessary, I do not see the same [logic present in vscode](https://github.com/microsoft/vscode/blob/7ecf23a2a2ca4905ca7c104de595632c8971d36a/src/vs/workbench/browser/parts/views/treeView.ts#L271-L275):

```ts
const oldEmpty = this._isEmpty;
this._isEmpty = children.length === 0;
if (oldEmpty !== this._isEmpty) {
    this._onDidChangeEmpty.fire();
}
```

Reply:
@vince-fugnitto this is the way i check that it is the root node.
In the vscode reference you shared the same condition exists with `instanceof`:
https://github.com/microsoft/vscode/blob/7ecf23a2a2ca4905ca7c104de595632c8971d36a/src/vs/workbench/browser/parts/views/treeView.ts#L270

As far as I can tell from debugging / source code, in theia all tree nodes are interface objects and not classes that we can do `instanceof` on. Is there more elegant way to identify root node?

Reply:
@amiramw I'm fine with your explanation thank you!